### PR TITLE
[ENH] Add ResNet PyTorch Implementation for Time Series Classification

### PR DESCRIPTION
**Pull request by amitsubhashchejara**
### Description
Implements ResNet (Residual Network) for time series classification using PyTorch backend, based on the architecture described in Wang et al. (2017).

**Part of #8699** - PyTorch migration initiative for deep learning networks.

### Current Status
🚧 **Draft PR - Seeking Feedback on PyTorch Testing Infrastructure**

### What's Included
- ✅ ResNetNetworkTorch class with residual blocks
- ✅ Use of `_safe_import` for soft PyTorch dependency
- ✅ Documentation and type hints
- ✅ Follows sktime coding conventions (pre-commit checks pass)

### Testing Status
- ⚠️ `check_estimator` tests encounter issues due to PyTorch testing infrastructure being in development
- ⚠️ Related to ongoing work in PRs #8842 (RNN PyTorch)

### Implementation Details

**Architecture (from Wang et al., 2017):**
- Three residual blocks with feature maps: [input_size → 64 → 128 → 128]
- Each block contains three convolutional layers with batch normalization
- Global average pooling followed by fully connected output layer

**Code Structure:**
- `ResNetNetworkTorch`: Main network class
- `ResidualBlock`: Individual residual block with skip connections
- Supports variable input sizes and number of classes
- Optional random seed for reproducibility

### Files Changed
```
sktime/networks/resnet/
├── __init__.py
├── _resnet_torch.py
```

### References
```bibtex
@inproceedings{wang2017time,
  title={Time series classification from scratch with deep neural networks: A strong baseline},
  author={Wang, Zhiguang and Yan, Weizhong and Oates, Tim},
  booktitle={2017 International joint conference on neural networks (IJCNN)},
  pages={1578--1585},
  year={2017},
  organization={IEEE}
}
```

### Checklist
- [x] Implementation follows paper specification
- [x] Code follows sktime conventions
- [x] Pre-commit checks pass
- [x] Documentation complete
- [ ] All tests passing (blocked by PyTorch testing infrastructure)
- [ ] Ready for merge (awaiting maintainer guidance)

### CODE REVIEW

**Comment by amitsubhashchejara** [comment]
> Thanks!
> 
> I suppose this is still work in progress, and this is still missing the classifier?

Yes, this is still work in progress. I just wanted to check if everything looks fine, I'll implement the classifier soon.

File: sktime/networks/resnet/_resnet_torch.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    ----------
+    input_size : int or tuple
+        If int, it represents the number of expected features in the input.
+        If tuple, it must be of the form (n_instances, n_dims, series_length)
+        where n_dims is the number of expected features in the input.
+    num_classes : int
+        Number of classes to predict.
+    random_state : int, optional (default = 0)
+        The random seed to use random activities.
+    activation : string, optional (default = "relu")
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py:35]: In sktime, generally there is a distinction made between `activation` function used in hidden layers and `activation` used in the final output layer, and both are specified separately.

It seems your code is not following this pattern.

File: sktime/networks/resnet/_resnet_torch.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    Zhiguang and Yan, Weizhong and Oates, Tim}, booktitle={2017
+    International joint conference on neural networks (IJCNN)}, pages={
+    1578--1585}, year={2017}, organization={IEEE} }
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["amitsubhashchejara"],
+        "python_version": ">=3.9",
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: Current supported python version is >= 3.10
Please check here: 

https://github.com/sktime/sktime/blob/bea23c642b646eb2116d000f37d92b63ac71e3e5/pyproject.toml#L32

File: sktime/networks/resnet/_resnet_torch.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    def __init__(
+        self,
+        input_size: int | tuple,
+        num_classes: int,
+        random_state: int = None,
+        activation: str = "relu",
+    ):
+        super().__init__()
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: `sktime` follows the pattern where class parameters are first stored in self and then base class is initialised. Please correct me if I'm wrong here, @fkiraly 

Your code does it the other way round. I understand you might have looked at the TensorFlow implementation of `ResNetNetwork` in `_resnet_tf.py`, but it seems its done incorrectly there as well. Correct way to do initialization here is:

```python
class myClass(baseClass):

    def __init__(
        param1,
        param2
    ):
        self.param1 = param1
        self.param2 = param2
        super.__init__()
        ...
```

**Review by RecreationalMath — CHANGES_REQUESTED** [review]
Good work, I've left some initial comments that should help you @amitsubhashchejara

I'll encourage you to implement the classifier in `PyTorch` which uses this network class so that you can utilise in-built tests to catch implementation errors.

Happy coding!

**Comment by amitsubhashchejara** [comment]
Thank you for the review and suggestions @RecreationalMath. I will make the changes.

**Comment by amitsubhashchejara** [comment]
Hi @RecreationalMath, I've worked on your suggestions and pushed some changes. Please check and let me know if I should proceed to implementing the estimator.
